include/dirac_quda.h

```diff
@@ -1568,8 +1644,43 @@ namespace quda {
     virtual bool hermitian() const { return true; } // normal op is always Hermitian
   };
 
+  /* Gloms onto a DiracOp and provides an operator() which applies its MdagMLocal */
+  class DiracMdagMLocal : public DiracMatrix
+  {
+
+  public:
+    DiracMdagMLocal(const Dirac &d) : DiracMatrix(d) {}
+    DiracMdagMLocal(const Dirac *d) : DiracMatrix(d) {}
+
+    void operator()(ColorSpinorField &out, const ColorSpinorField &in) const { dirac->MdagMLocal(out, in); }
+
+    void operator()(ColorSpinorField &out, const ColorSpinorField &in, ColorSpinorField &tmp) const
+    {
+      dirac->tmp1 = &tmp;
+      dirac->MdagMLocal(out, in);
+      if (shift != 0.0) blas::axpy(shift, const_cast<ColorSpinorField &>(in), out);
```

Comment:
Any reason there isn't a fused `MdagMLocal`+`axpy`?

Reply:
@weinbe2 Do you mean a fused kernel for `MdagMLocal` + `axpy`? The `if (shift != 0.0) blas::axpy(...)` pattern is used in the other methods (`MdagM`/`M`) as well: I am just replicating here. In principle a fused kernel is certainly better but I would push that to the following boundary-condition-cleaning PR that's on my list.

Reply:
This is a non issue.  I introduced this as an initial hack for some variant of Hasenbusch preconditioning, for plugging into BQCD IIRC.  No need for a fused kernel for this.